packages/git/src/index.ts

```diff
@@ -26,6 +26,16 @@ async function commit(message: string, cwd: string) {
   return gitCmd.code === 0;
 }
 
+async function getAllTags(cwd: string): Promise<string[]> {
+  const gitCmd = await spawn("git", ["tag"], { cwd });
+  if (gitCmd.code === 0) {
+    const tags = gitCmd.stdout.toString().split("\n");
```

Comment:
With my git version/config the `stdout` contains a trailing newline - so to be more precise we'd need this:
```suggestion
    const tags = gitCmd.stdout.toString().trim().split("\n");
```

Reply:
I'm also wondering if this would work on WIndows - is the line separator unified there in this context?

Reply:
I booted up my windows laptop (despite a bsod!) and tested this out. Everything worked, although I ended up adding a `.filter(Boolean)` because it was extracting an empty string as a tag, even after I added the `.trim()`.

Here's the code I ran on my windows machine as a test:

```js
const spawn = require('spawndamnit');

spawn("git", ["tag"], { cwd: process.cwd() }).then(gitCmd => {
  const tags = gitCmd.stdout.toString().trim().split("\n").filter(Boolean);
  console.log("tags", tags);
})
```

I ran the test code in a repo with no tags, with one tag, and with two tags. It worked in all cases. Regarding windows new line characters, I'm not sure why the `"\n"` works there but I think it's because a lot of Windows programs use `\n\r` in their code, so that code looking for normal new lines or carriage returns all work. It also might be possible that NodeJS's spawn command does some cross-platform stuff when converting the stdout to a node string. Either way, it seems to work well on Windows.

Reply:
Here's the resulting output:

```sh
chars [
  118, 49, 46, 48, 46, 48,
  10, 118, 49, 46, 48, 46,
  49, 10
]
```

Reply:
That array is an equivalent of `'v1.0.0\nv1.0.1\n'`, with no hidden character of anything like that. So a single `.trim()` should already handle that:
```js
[118, 49, 46, 48, 46, 48, 10, 118, 49, 46, 48, 46, 49, 10 ].map(s => String.fromCharCode(s)).join('').trim().split("\n") // ['v1.0.0', 'v1.0.1']
```

I would prefer to avoid that extra `.filter(Boolean)` unless we can write a test that would fail without it.